hack/update-vendor-licenses.sh

```diff
@@ -186,8 +186,60 @@ if [ -f "${LICENSE_ROOT}/LICENSE" ]; then
   mv "${TMP_LICENSE_FILE}" "${TMP_LICENSES_DIR}/LICENSE"
 fi
 
+sorted_modules=$("$go" list -m -json all | "$jq" -r .Path | sort -f)
+
+# is_module checks whether the named package is a module
+is_module() {
+  package="$1"
+  for MODULE in $sorted_modules; do
+    if [ "${MODULE}" == "${package}" ]; then
+      return 0
+    fi
+  done
+  return 1
+}
+
+# only_contains_submodules will check whether a package is empty of files
+# and its subdirectories are all modules, which indicates we do not need
+# a LICENSE for this module as it is an indirect transitive dependency.
+only_contains_submodules () {
+  package="$1"
+  if [[ -z "$(find "${DEPS_DIR}/${package}/" -mindepth 1 -maxdepth 1 -type f)" ]]; then
+    # If the package does not contain files, check whether all subdirectories
+    # only contain modules.
+    while read -d "" -r SUBDIR; do
+      # Because we have a trailing '/' on the 'find' command's argument, we
+      # must strip off two '/' from the end of the package path before the
+      # sub-packages name.
```

Comment:
The reason for actually ending the `find` command with a `/` (meaning we have to strip two of them off) is explained here: https://github.com/kubernetes/kubernetes/pull/92729